src/Modal/ModalLayer.jsx

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/*  TODO: The first of these two disabled linters is okay.
+Focus lock is handling the keyboard for us. The second I'm not sure */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FocusOn } from 'react-focus-on';
+import Portal from './Portal';
+import { ModalContextProvider } from './ModalContext';
+
+const ModalBackdrop = ({ onClick }) => (
+  <div className="pgn__modal-backdrop" onClick={onClick} />
+);
+
+ModalBackdrop.propTypes = {
+  onClick: PropTypes.func,
+};
+
+ModalBackdrop.defaultProps = {
+  onClick: undefined,
+};
+
+const ModalContentContainer = ({ children }) => (
+  <div className="pgn__modal-content-container">{children}</div>
+);
+
+ModalContentContainer.propTypes = {
+  children: PropTypes.node,
+};
+
+ModalContentContainer.defaultProps = {
+  children: null,
+};
+
+/**
+ * The ModalLayer should be used for any component that wishes to engage the user
+ * in a "mode" where a layer on top of the application is interactive while the
+ * rest of the application is made non-interactive. The assumption made by this
+ * component is that if a modal object is visible then it is "enabled"
+ */
+const ModalLayer = ({
+  children, close, isOpen, isBlocking,
+}) => {
+  if (!isOpen) {
+    return null;
+  }
+
+  return (
+    <ModalContextProvider close={close} isOpen={isOpen} isBlocking={isBlocking}>
+      <Portal>
+        <FocusOn
+          scrollLock
+          enabled={isOpen}
+          onEscapeKey={close}
+          onClickOutside={!isBlocking && close}
+          className="pgn__modal-layer"
+        >
+          <ModalContentContainer>
+            <ModalBackdrop onClick={!isBlocking && close} />
```

Comment:
Is this the bit that results in the jsx-a11y eslint error? 

This `onClick` prop gets passed to a `div` element but a `div` is not inherently interactive and would typically require a `role` attribute to be added to indicate that this element is actually interactive, e.g. `role="link"` or `role="button"`.

![image](https://user-images.githubusercontent.com/2828721/107819245-d85c5480-6d46-11eb-82b1-23cc24d15835.png)

https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-static-element-interactions.md#case-the-event-handler-is-only-being-used-to-capture-bubbled-events

I think it's ok to disable this rule but would recommend moving the eslint-disable comment above the line which causes it with an explanation to reduce confusion.

Reply:
👍 moved